### PR TITLE
fix: removed lucide aliases

### PR DESCRIPTION
**Pull request by jguddas**
#53483 introduced support for lucide aliases in next.js, to not have to maintain this mapping in two places lucide will have a file for every alias.

https://github.com/lucide-icons/lucide/pull/1486

### CODE REVIEW

File: packages/next/src/server/config.ts [hunk 1]
```diff
@@ -684,35 +684,6 @@ function assignDefaults(
...
-        '(SquareKanban|LucideSquareKanban|SquareKanbanIcon)':
-          'modularize-import-loader?name={{ member }}&from=default&as=default&join=../esm/icons/kanban-square!lucide-react',
-        '(Edit3|LucideEdit3|Edit3Icon)':
-          'modularize-import-loader?name={{ member }}&from=default&as=default&join=../esm/icons/pen-line!lucide-react',
-        '(Edit|LucideEdit|EditIcon|PenBox|LucidePenBox|PenBoxIcon)':
-          'modularize-import-loader?name={{ member }}&from=default&as=default&join=../esm/icons/pen-square!lucide-react',
-        '(Edit2|LucideEdit2|Edit2Icon)':
-          'modularize-import-loader?name={{ member }}&from=default&as=default&join=../esm/icons/pen!lucide-react',
-        '(Stars|LucideStars|StarsIcon)':
-          'modularize-import-loader?name={{ member }}&from=default&as=default&join=../esm/icons/sparkles!lucide-react',
```
Comment (jguddas) [inline packages/next/src/server/config.ts:713]: Now there is a `/esm/icons/sparkles.js` file containining a simple reexport from `sparkles`.

```js
/**
 * lucide-react v0.267.0 - ISC
 */

export { default } from './sparkles.js';
//# sourceMappingURL=stars.js.map
```

https://www.npmjs.com/package/lucide-react?activeTab=code

**Comment by jd-carroll** [comment]
@ztanner we are really excited to use #54016 but will not be able to use the latest due to this issue.  Any chance we can get this PR merged soon?

**Review by shuding — CHANGES_REQUESTED** [review]
Can we fix the wrong transform rule instead of removing all of them? This breaks the optimization for older versions of Lucide React.

**Comment by jguddas** [comment]
> This breaks the optimization for older versions of Lucide React.

@shuding old versions of lucide are already broken, since they use `./icons` rather than `../esm/icons`.

There are only very view versions that have both `../esm/icons` but don't have the aliases i.e. `../esm/icons/stars.js` that @ericfennis and I added so we don't have to maintain them in both places.

Keeping those aliases here will break as soon as we add a `starts` icon to lucide for example.

If there are 250 or 253 prerelease versions of lucide that don't work is IMO okay considering this problem.

**Pull request closed without merging**